scripts/pylib/twister/twisterlib/size_calc.py

```diff
@@ -224,3 +275,174 @@ def _calculate_sizes(self):
                                   "size": size, "virt_addr": virt_addr,
                                   "type": stype, "recognized": recognized})
+
+    def _analyze_elf_file(self) -> None:
+        self._check_elf_file()
+        self._check_is_xip()
+        self._get_info_elf_sections()
+
+    def _get_buildlog_file_content(self) -> typing.List[str]:
+        """Get content of the build.log file.
+
+        @return Content of the build.log file (list[str])
+        """
+        if os.path.exists(path=self.buildlog_filename) and self.buildlog_filename.endswith("build.log"):
+            with open(file=self.buildlog_filename, mode='r') as file:
+                file_content = file.readlines()
+        else:
+            logger.error("Incorrect build.log file to analyze footprints.")
+            file_content = []
+        return file_content
+
+    def _find_offset_of_last_pattern_occurrence(self, file_content: typing.List[str]) -> int:
+        """Find the offset from which the information about the memory footprint is read.
+
+        @param file_content (list[str]) Content of build.log.
+        @return: Offset with information about the memory footprint (int)
+        """
+        result = -1
+        if len(file_content) == 0:
+            logger.warning("Build.log file is empty.")
+        else:
+            # Check the file in reverse order.
+            for idx, line in enumerate(reversed(file_content)):
+                # Condition is fulfilled if the pattern matches with the start of the line.
+                if re.match(pattern=self.PATTERN_SEARCHED_LINE, string=line):
+                    result = idx + 1
+                    break
+        # If the file does not contain information about memory footprint, the warning is raised.
+        if result == -1:
+            logger.warning(msg="Information about memory footprint for this test configuration is not found.")
+        return result
+
+    def _get_lines_with_footprint(self, start_offset: int, file_content: typing.List[str]) -> typing.List[str]:
+        """Get lines from the file with a memory footprint.
+
+        @param start_offset (int) Offset with the first line of the information about memory footprint.
+        @param file_content (list[str]) Content of the build.log file.
+        @return: Lines with information about memory footprint (list[str])
+        """
+        if len(file_content) == 0:
+            logging.warning(msg="Build.log file is empty.")
+            result = []
+        else:
+            if start_offset > len(file_content) or start_offset <= 0:
+                info_line_idx_start = len(file_content) - 1
+            else:
+                info_line_idx_start = len(file_content) - start_offset
+
+            info_line_idx_stop = info_line_idx_start + self.USEFUL_LINES_AMOUNT
+            if info_line_idx_stop > len(file_content):
+                info_line_idx_stop = len(file_content) - 1
```

Comment:
Should be without -1:
```python
info_line_idx_stop = len(file_content)
```
Sorry, for this - it was my suggestion in previous comment and there was a bug. In Python it is possible to use in index range number like `len(mylist)`, for example:
```python
mylist = [1, 2, 3, 4]
print(mylist[2:len(mylist)])
# what is the same as:
print(mylist[2:4])
# or simpler:
print(mylist[2:])
```

Reply:
Done